backend/dataall/api/Objects/Dataset/resolvers.py

```diff
@@ -212,6 +228,9 @@ def get_dataset_stewards_group(context, source: models.Dataset, **kwargs):
 
 def update_dataset(context, source, datasetUri: str = None, input: dict = None):
     with context.engine.scoped_session() as session:
+        dataset = Dataset.get_dataset_by_uri(session, datasetUri)
+        environment = Environment.get_environment_by_uri(session, dataset.environmentUri)
+        check_dataset_account(environment=environment)
```

Comment:
Minor: looks like we can create `check_dataset_account(dataset_uri=uri)` and move  `get_environment_by_uri` into this method.

Reply:
In the update_dataset method yes, but in the create method it is not possible. When we create we are checking the environment account before the dataset object exists